code/lib/cli/src/detect.ts

```diff
@@ -109,28 +110,47 @@ export function detectFrameworkPreset(
  *
  * @returns CoreBuilder
  */
-export function detectBuilder(packageManager: JsPackageManager, projectType: ProjectType) {
+export async function detectBuilder(packageManager: JsPackageManager, projectType: ProjectType) {
   const viteConfig = findUp.sync(viteConfigFiles);
-  if (viteConfig) {
+  const webpackConfig = findUp.sync(webpackConfigFiles);
+  const dependencies = await packageManager.getAllDependencies();
+
+  if (viteConfig || (dependencies['vite'] && dependencies['webpack'] === undefined)) {
     paddedLog('Detected Vite project. Setting builder to Vite');
     return CoreBuilder.Vite;
   }
 
-  if (detectWebpack(packageManager)) {
+  // REWORK
+  if (webpackConfig || (dependencies['webpack'] && dependencies['vite'] !== undefined)) {
     paddedLog('Detected webpack project. Setting builder to webpack');
     return CoreBuilder.Webpack5;
   }
 
   // Fallback to Vite or Webpack based on project type
   switch (projectType) {
-    case ProjectType.SVELTE:
-    case ProjectType.SVELTEKIT:
     case ProjectType.VUE:
     case ProjectType.VUE3:
     case ProjectType.SFC_VUE:
       return CoreBuilder.Vite;
-    default:
+    case ProjectType.REACT_SCRIPTS:
```

Comment:
Mentioning some webpack5-only projects, where we know that they are not currently working with Vite.